### PR TITLE
docs: improve community pages

### PR DESCRIPTION
**Pull request by phonzammi**
I just created a PR to update the docs and mention the new `vike-react-styled-components` extension.
Please take a look and review this PR.

Which release docs do we need to update ?

### CODE REVIEW

**Comment by brillout** [comment]
The latest unpublished one, so the same than the last PR

File: docs/pages/styled-components/+Page.mdx [hunk 1]
```diff
@@ -3,13 +3,23 @@ import { Link } from '@brillout/docpress'
...
+
+## Manual integration
+
+When using styled-components, you need to collect the page's styles upon SSR:
+- [styled-components > Server Side Rendering](https://styled-components.com/docs/advanced#server-side-rendering)
 - <Link href="/css-in-js" doNotInferSectionTitle />
 
 You can follow along on <a href="https://www.alexrabin.com/blog/vite-ssr-styled-components">this tutorial</a> (vite-plugin-ssr was the [previous name of Vike](https://vite-plugin-ssr.com/vike)) on how to add styled-components to your Vike app.
 
+If you use <Link href="/vike-react">`vike-react`</Link>, you can manually integrate styled-components using a combination of these:
```
Comment (brillout) [inline docs/pages/styled-components/+Page.mdx:16]:
```suggestion
If you use <Link href="/vike-react">`vike-react`</Link>, you can manually integrate styled-components by using a combination of these:
```

File: docs/pages/styled-components/+Page.mdx [hunk 2]
```diff
@@ -3,13 +3,23 @@ import { Link } from '@brillout/docpress'
...
 <CommunityNote />
 
-When using [styled-components](https://styled-components.com), you need to collect the page's styles upon SSR:
-- <a href="https://styled-components.com/docs/advanced#server-side-rendering">styled-components docs &gt; Server Side Rendering</a>
+To use [styled-components](https://styled-components/), you can use [`vike-react-styled-components`](https://github.com/vikejs/vike-react/tree/main/packages/vike-react-styled-components#readme) for an automatic integration with Vike.
+
+## Manual integration
+
+When using styled-components, you need to collect the page's styles upon SSR:
+- [styled-components > Server Side Rendering](https://styled-components.com/docs/advanced#server-side-rendering)
```
Comment (brillout) [inline docs/pages/styled-components/+Page.mdx:11]: Let's create a `## See also` and put the link in there as well.

File: docs/pages/styled-components/+Page.mdx [hunk 3]
```diff
@@ -3,13 +3,23 @@ import { Link } from '@brillout/docpress'
...
 - <Link href="/css-in-js" doNotInferSectionTitle />
 
 You can follow along on <a href="https://www.alexrabin.com/blog/vite-ssr-styled-components">this tutorial</a> (vite-plugin-ssr was the [previous name of Vike](https://vite-plugin-ssr.com/vike)) on how to add styled-components to your Vike app.
 
+If you use <Link href="/vike-react">`vike-react`</Link>, you can manually integrate styled-components using a combination of these:
+- <Link href="/onBeforeRenderHtml" doNotInferSectionTitle />.
```
Comment (brillout) [inline docs/pages/styled-components/+Page.mdx:17]: Let's remove the trailing dots.

File: docs/pages/styled-components/+Page.mdx [hunk 4]
```diff
@@ -3,13 +3,23 @@ import { Link } from '@brillout/docpress'
 
 <CommunityNote />
 
-When using [styled-components](https://styled-components.com), you need to collect the page's styles upon SSR:
-- <a href="https://styled-components.com/docs/advanced#server-side-rendering">styled-components docs &gt; Server Side Rendering</a>
+To use [styled-components](https://styled-components/), you can use [`vike-react-styled-components`](https://github.com/vikejs/vike-react/tree/main/packages/vike-react-styled-components#readme) for an automatic integration with Vike.
```
[the remark this answers is not in the document]
Comment (phonzammi) [reply docs/pages/styled-components/+Page.mdx:6]: Does `brillout/docpress` export the `usePageContext()` hook?

[the remark this answers is not in the document]
Comment (brillout) [reply docs/pages/styled-components/+Page.mdx:6]: If it doesn't already we can update docpress to do it.

[the remark this answers is not in the document]
Comment (phonzammi) [reply docs/pages/styled-components/+Page.mdx]: No, it doesn't.

[the remark this answers is not in the document]
Comment (brillout) [reply docs/pages/styled-components/+Page.mdx]: PR welcome to `@brillout/docpresss`.

[the remark this answers is not in the document]
Comment (brillout) [reply docs/pages/styled-components/+Page.mdx]: Or I can quickly do it, let me check.

[the remark this answers is not in the document]
Comment (brillout) [reply docs/pages/styled-components/+Page.mdx]: Released in `@brillout/docpress@0.9.3`

[the remark this answers is not in the document]
Comment (phonzammi) [reply docs/pages/styled-components/+Page.mdx]: Thank you, I'll push a commit to update the docpress directly in this PR

[the remark this answers is not in the document]
Comment (phonzammi) [reply docs/pages/styled-components/+Page.mdx]: It seems that `pageContext` is `undefined` on the client side

[the remark this answers is not in the document]
Comment (phonzammi) [reply docs/pages/styled-components/+Page.mdx]: I've pushed the failing commit, please try it and open the `/styled-components` page

[the remark this answers is not in the document]
Comment (brillout) [reply docs/pages/styled-components/+Page.mdx]: Fixed & released in `0.9.4`.

[the remark this answers is not in the document]
Comment (phonzammi) [reply docs/pages/styled-components/+Page.mdx]: Please take a look at this commit : https://github.com/vikejs/vike/pull/1999/commits/c12e12fdfbd5319b2096ec196259b52f33e6743e

What do you think ?

[the remark this answers is not in the document]
Comment (brillout) [reply docs/pages/styled-components/+Page.mdx]: LGTM. I think we can remove the tool name prop althogether

[the remark this answers is not in the document]
Comment (phonzammi) [reply docs/pages/styled-components/+Page.mdx]: Nice, should we make the `url` prop required and use `assert()`?

[the remark this answers is not in the document]
Comment (phonzammi) [reply docs/pages/styled-components/+Page.mdx]: Okay, now I'll update all pages that are using `<CommunityNote>` component

**Comment by phonzammi** [comment]
> I'll update all pages that are using `<CommunityNote>` component

Done : https://github.com/vikejs/vike/pull/1999/commits/e62b96094e25d11bf7657411a71d5dab5ebaeda6

**Comment by brillout** [comment]
Is the PR ready for review?

**Comment by phonzammi** [comment]
> Is the PR ready for review?

Yes, please

File: docs/components/CommunityNote.tsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
+export { CommunityNote }
+
+import React from 'react'
+import { assert, Contribution, usePageContext } from '@brillout/docpress'
+import { getEditLink } from '../utils'
+
+function CommunityNote({ url }: { url: string }) {
+  assert(url, 'The `url` prop are required and cannot be undefined')
```
Comment (brillout) [inline docs/components/CommunityNote.tsx:8]:
```suggestion
  assert(url, 'The `url` prop is required')
```

File: docs/components/CommunityNote.tsx [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  const pageContext = usePageContext()
+  return (
+    <>
+      Community-led documentation about using <a href={url || '#'}>{pageContext.pageTitle}</a> with Vike.
```
Comment (brillout) [inline docs/components/CommunityNote.tsx:12]:
```suggestion
      Community-led documentation about using <a href={url}>{pageContext.pageTitle}</a> with Vike.
```

File: docs/components/CommunityNote.tsx [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+      <Contribution>
+        This page may contain outdated information,{' '}
+        <a href={getEditLink(pageContext.urlPathname)} target="_blank">
```
Comment (brillout) [inline docs/components/CommunityNote.tsx:15]: Let's define `getEditLink()` in this file.

File: docs/pages/bootstrap/+Page.mdx [hunk 8]
```diff
@@ -1,6 +1,6 @@
 import { CommunityNote } from '../../components'
 
-<CommunityNote />
+<CommunityNote url="https://getbootstrap.com" />
 
 Using Vike with [Bootstrap](https://getbootstrap.com).
```
Comment (brillout) [inline docs/pages/bootstrap/+Page.mdx:5]:
```suggestion
```

File: docs/pages/MDXEditor/+Page.mdx [hunk 9]
```diff
@@ -1,7 +1,7 @@
 import { Link } from '@brillout/docpress'
 import { CommunityNote } from '../../components'
 
-<CommunityNote />
+<CommunityNote url="https://mdxeditor.dev" />
 
 To use [MDXEditor](https://github.com/mdx-editor/editor) make sure to <Link href="/clientOnly">load it only on the client-side</Link>.
```
Comment (brillout) [inline docs/pages/MDXEditor/+Page.mdx:6]:
```suggestion
To use MDXEditor make sure to <Link href="/clientOnly">load it only on the client-side</Link>.
```

File: docs/pages/compiled/+Page.mdx [hunk 10]
```diff
@@ -1,5 +1,5 @@
 import { UseBati, CommunityNote } from '../../components'
 
-<CommunityNote />
+<CommunityNote url="https://compiledcssinjs.com" />
 
 <UseBati>[Compiled](https://compiledcssinjs.com)</UseBati>
```
Comment (brillout) [inline docs/pages/compiled/+Page.mdx:5]:
```suggestion
<UseBati>Compiled</UseBati>
```

It's a bit confusing to repeat the link in the next sentence. Let's avoid this everywhere.

File: docs/pages/effector/+Page.mdx [hunk 11]
```diff
@@ -1,7 +1,7 @@
 import { Link } from '@brillout/docpress'
 import { CommunityNote } from '../../components'
 
-<CommunityNote />
+<CommunityNote url="https://effector.dev" />
 
 You can use Vike with [Effector](https://effector.dev).
```
Comment (brillout) [inline docs/pages/effector/+Page.mdx:6]:
```suggestion
```

This is now superfluous, let's remove it everywhere.

File: docs/pages/tanstack-query/+Page.mdx [hunk 12]
```diff
@@ -1,14 +1,17 @@
...
 import { Example, UiFrameworkExtension, CommunityNote } from '../../components'
 
-<CommunityNote />
+<CommunityNote url="https://tanstack.com/query" />
 
 You can use Vike with [TanStack Query](https://tanstack.com/query).
 
 > Instead of manually integrating TanStack Query yourself, you can use:
 > - [`vike-react-query`](https://github.com/vikejs/vike-react/tree/main/packages/vike-react-query)
+> - [`vike-solid-query`](https://github.com/vikejs/vike-solid/tree/main/packages/vike-solid-query)
```
Comment (brillout) [inline docs/pages/tanstack-query/+Page.mdx:10]: :100:

**Comment by phonzammi** [comment]
I've made all the requested changes. Let me know if I missed anything.

**Comment by brillout** [comment]
There are still a lot of duplicated links. (I've removed [some](https://github.com/vikejs/vike/pull/1999/commits/9cb19a5501adcc1e3aefe8d3315fbd536ae8caa4) but there are still some left.)

File: docs/pages/sass/+Page.mdx [hunk 13]
```diff
@@ -1,6 +1,6 @@
 import { Example, CommunityNote } from '../../components'
 
-<CommunityNote />
+<CommunityNote url="https://sass-lang.com" />
 
 For using [Sass](https://sass-lang.com)/[Less](https://lesscss.org)/[Stylus](https://stylus-lang.com), all you have to do is to install the CSS pre-processor you want to use: Vite will then automatically integrate it. See [Vite Docs - CSS pre-processors](https://vitejs.dev/guide/features.html#css-pre-processors).
```
Comment (brillout) [inline docs/pages/sass/+Page.mdx:5]: Duplicated link. Intervention++. (It's much slower for me to review when I have to review details, that's why I'd be great if we minimize these kind of interventions.)

Comment (brillout) [reply docs/pages/sass/+Page.mdx:5]: Ah, I see why you kept it. Nevermind.

File: docs/pages/nginx/+Page.mdx [hunk 14]
```diff
@@ -1,7 +1,7 @@
 import { Link } from '@brillout/docpress'
 import { CommunityNote } from '../../components'
 
-<CommunityNote />
+<CommunityNote url="https://nginx.org" />
 
 Many users are using Vike with [Nginx](https://nginx.org/).
```
Comment (brillout) [inline docs/pages/nginx/+Page.mdx:6]: Duplicated link. Intervention++

File: docs/pages/hapi/+Page.mdx [hunk 15]
```diff
@@ -1,9 +1,9 @@
 import { Link } from '@brillout/docpress'
 import { CommunityNote } from '../../components'
 
-<CommunityNote />
+<CommunityNote url="https://hapi.dev" />
 
-We can use Vike as well with [hapi](https://hapi.dev/).
+We can use Vike as well with hapi.
```
Comment (brillout) [inline docs/pages/hapi/+Page.mdx:6]:
```suggestion
```

File: docs/pages/hapi/+Page.mdx [hunk 16]
```diff
@@ -1,9 +1,9 @@
...
 
 > hapi is an [Express.js](https://expressjs.com/) alternative.
```
Comment (brillout) [inline docs/pages/hapi/+Page.mdx:8]:
```suggestion
```

File: docs/pages/apollo-graphql/+Page.mdx [hunk 17]
```diff
@@ -21,3 +21,8 @@ Examples of integrating Apollo without using <UiFrameworkExtension name />:
 - <Example timestamp="2024.07" repo="brillout/vike-apollo-vue" /> - With Vue
 
 > You can also have a look at [the source code of `vike-react-apollo`](https://github.com/vikejs/vike-react/tree/main/packages/vike-react-apollo).
+
+## See also
```
Comment (brillout) [inline docs/pages/apollo-graphql/+Page.mdx:25]: :+1: :sparkles:

File: docs/pages/cloudflare-pages/+Page.mdx [hunk 18]
```diff
@@ -11,6 +10,8 @@ It will deploy prerendered routes as static files, and dynamic routes as [Pages
 
 > Instead of using `vike-cloudflare` you can manually integrate Cloudflare Pages yourself, see <Link href="/cloudflare-workers#cloudflare-pages" doNotInferSectionTitle={true} />.
 
+<UseBati>Cloudflare Pages with `vike-cloudflare`</UseBati>
```
Comment (brillout) [inline docs/pages/cloudflare-pages/+Page.mdx:13]: Good catch :dizzy:

**Comment by phonzammi** [comment]
@brillout, it's my pleasure too! I've learned a lot, thanks to you and the Vike team

> Idea: how about we define a new prop `<CummunityNote hasExtension>`?
Believe it or not, I thought something similar, just with an additional component. I like your approach more. 💯

**Comment by brillout** [comment]
:100: Let's do it!

(The idea of making it a prop instead of a separate component occurred to me while writing my comment :smirk:)

**Pull request merged**